Documentation/ExtensionArchitecture/Extbase/Reference/Validation/Validators/Index.rst

```diff
@@ -0,0 +1,624 @@
+=======================================
+
+This document lists all built-in Extbase validators available in TYPO3 13,
+along with example usage for each using PHP attributes.
+
+..  seealso::
+    *   `Using validation for Extbase models and controllers <https://docs.typo3.org/permalink/t3coreapi:extbase-validation>`_
+    *   `Custom Extbase validator implementation <https://docs.typo3.org/permalink/t3coreapi:extbase-domain-validator>`_
+
+..  contents:: Validators in Extbase
+
+..  _extbase-validator-alpanumeric:
+
+AlphanumericValidator
+=====================
+
+Ensures a value contains only letters and numbers.
+
+..  code-block:: php
+
+    #[Validate(['validator' => 'Alphanumeric'])]
+    protected string $username;
+
+..  _extbase-validator-boolean:
+
+
+BooleanValidator
+================
+
+Checks if a value matches a specific boolean value (`true` or `false`).
+
+This validator is useful when you want to enforce that a property explicitly
+evaluates to either `true` or `false`, such as for checkboxes in forms.
+
+By default, it does not enforce any specific value unless the `is` option is
+set.
+
+**Options:**
+-   `is`: Accepts `'true'`, `'1'`, `'false'`, `'0'`, or `''`. Values are
+  converted internally to boolean `true` or `false`.
+
+Ensure that a value is a boolean (no strict check, default behavior):
+
+..  code-block:: php
+
+    #[Validate(['validator' => 'Boolean'])]
+    protected $isActive;
+
+Require that a value must be `true` (e.g. checkbox must be checked):
+
+..  code-block:: php
+
+    #[Validate(['validator' => 'Boolean', 'options' => ['is' => 'false']])]
+    protected bool $termsAccepted;
+
+Require that a value must be `false`:
+
+..  code-block:: php
+
+    #[Validate(['validator' => 'Boolean', 'options' => ['is' => 'false']])]
+    protected bool $isBlocked;
+
+..  _extbase-validator-collection:
+
+CollectionValidator
+===================
+
+The `CollectionValidator` is a built-in Extbase validator for validating arrays
+or collections, such as arrays of DTOs or `ObjectStorage<T>` elements.
+
+It allows you to apply validation rules to **each individual item** in a
+collection. The validation is recursive: every item is passed through the
+validator you specify.
+
+`elementValidator`
+    The name or class of a validator that should be applied
+    to each item in the collection (e.g. `'NotEmpty'`, `'EmailAddress'`).
+`elementType`
+    The class name of the collection's element type. All
+    registered validators for that type will be applied to each item.
+
+You must provide **either** `elementValidator` or `elementType`.
+
+..  include:: _NoExamples.rst.txt
+
+Use cases:
+
+-   Validating dynamic or repeatable form fields (e.g. multiple answers)
+-   Validating input arrays from multi-select fields or checkboxes
+-   Validating each related object in an `ObjectStorage` property
+
+..  note::
+    Validation will be skipped if neither `elementValidator` nor
+    `elementType` is set.
+
```

Comment:
Can we have a snippet here, too?

Reply:
`.. include:: _NoExamples.rst.txt`

So no unless you can provide one ;.)